### PR TITLE
Update readme and changelog for app open lifecycle changes.

### PR DESCRIPTION
**Pull request by jjliu15**
## Description

Follow up to https://github.com/googleads/googleads-mobile-flutter/pull/425. Updates the changelog and readme.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/app_open_example/README.md [hunk 1]
```diff
@@ -1,3 +1,320 @@
...
+          print('AppOpenAd failed to load: $error');
+          // Handle the error.
+        },
+      ),
+    );
+  }
+}
+```
+
+## Show the ad and handle fullscreen callbacks
```
Comment (luo86) [inline packages/app_open_example/README.md:86]: Add one #?  should it be in the same level as "Load an ad"?

File: packages/app_open_example/README.md [hunk 2]
```diff
@@ -1,3 +1,320 @@
...
+      },
+    );
+  }
+}
+```
+
+If a user returns to your app after having left it by clicking on an app open
+ad, make sure they're not presented with another app open ad.
+
+## Listen for app foregrounding events
```
Comment (luo86) [inline packages/app_open_example/README.md:132]: Add one #?  should it be in the same level as "Load an ad"?

File: packages/google_mobile_ads/CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,11 @@
           }
       }
       ```
+* Adds a workaround for issue [#404](https://github.com/googleads/googleads-mobile-flutter/issues/404)
```
Comment (luo86) [inline packages/google_mobile_ads/CHANGELOG.md:14]: Should / Could we change "Adds a workaround for issue" to "Fixes the issue"?
Workaround usually means not a final or good solution and will be changed with a good solution later. Unless that's the case that we will change it later soon.

**Review by luo86 — APPROVED** [review]

File: packages/app_open_example/README.md (hunk 1 above)
Comment (jjliu15) [reply packages/app_open_example/README.md:86]: Done

**Pull request merged**